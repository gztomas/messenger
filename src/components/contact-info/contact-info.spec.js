import React from 'react';
import { shallow } from 'enzyme';
import ContactInfo from './contact-info';

describe('ContactInfo', () => {
  let props;

  beforeEach(() => {
    props = {
      picture: 'http://cdn.com/picture.jpg',
      name: 'Victor',
      status: 'online',
      isTyping: false,
    };
  });

  it('should render properly', () => {
    const wrapper = shallow(<ContactInfo {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show the name', () => {
    const wrapper = shallow(<ContactInfo {...props} />);
    const title = wrapper.find('h4');
    expect(title.text()).toBe('Victor');
  });

  it('should show the status', () => {
    const wrapper = shallow(<ContactInfo {...props} />);
    const details = wrapper.find('span');
    expect(details.text()).toBe('online');
  });

  it('should show when typing', () => {
    props.isTyping = true;
    const wrapper = shallow(<ContactInfo {...props} />);
    const details = wrapper.find('span');
    expect(details.text()).toBe('typing...');
  });
});
