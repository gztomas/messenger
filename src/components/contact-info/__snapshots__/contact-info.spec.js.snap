// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo should render properly 1`] = `
<styled__StyledContainer>
  <Avatar
    shape="circle"
    size="large"
    src="http://cdn.com/picture.jpg"
  />
  <styled__StyledDetails>
    <h4>
      Victor
    </h4>
    <span>
      online
    </span>
  </styled__StyledDetails>
</styled__StyledContainer>
`;
